File: services/ms-cliente/src/main/java/airportsystem/mscliente/repository/TransacaoMilhasRepository.java
package airportsystem.mscliente.repository;

import airportsystem.mscliente.model.TransacaoMilhas;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TransacaoMilhasRepository extends JpaRepository<TransacaoMilhas, Long> {
    List<TransacaoMilhas> findByClienteCodigoOrderByDataHoraDesc(String codigo);
}

File: services/ms-cliente/airport-system.ms-cliente/src/main/kotlin/com/tads/airport_system/ms_cliente/repository/TransacaoMilhasRepository.kt
package com.tads.airport_system.ms_cliente.repository

import com.tads.airport_system.ms_cliente.model.TransacaoMilhas
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TransacaoMilhasRepository : JpaRepository<TransacaoMilhas, Long> {
    fun findByClienteCpfOrderByDataHoraDesc(cpf: String): List<TransacaoMilhas>
} 